src/ledger/LedgerTxnTrustLineSQL.cpp

```diff
@@ -414,7 +414,11 @@ class BulkLoadTrustLinesOperation
         for (auto const& k : keys)
         {
             releaseAssert(k.type() == TRUSTLINE);
-            releaseAssert(k.trustLine().asset.type() != ASSET_TYPE_NATIVE);
+            if (k.trustLine().asset.type() == ASSET_TYPE_NATIVE)
```

Comment:
in the other bulk* functions we use `validateTrustLineKey` instead

Reply:
I didn't use `validateTrustLineKey` here because that would technically be a protocol change. For example, we would return an `txINTERNAL_ERROR` when trying to bulk load an invalid asset, which we don't do at the moment.

Reply:
we call it already in `LedgerTxnRoot::Impl::loadTrustLine` so right now it's more a footgun that bulk load doesn't have the same invariants than point load

Reply:
So do you propose that @sisuresh should use `validateTrustLineKey` but gate it on protocol version? Otherwise it's a protocol change, because we don't necessarily load everything we prefetch.

Reply:
So I tried to use `validateTrustLineKey` as a protocol change, but it looks like we can actually "prefetch" the trustline for the issuer in most operations that use trustlines. Ex. https://github.com/stellar/stellar-core/blob/fb2948e02579257433432ffa7aab173ccbd5aaa1/src/transactions/PathPaymentOpFrameBase.cpp#L37

Reply:
@sisuresh Independent of what we decide to do in this thread, please open an issue for us to fix that later.

Reply:
agree @jonjove . let's discuss other problems elsewhere